auth/internal/transport/cert/secureconnect_cert.go

```diff
@@ -62,11 +62,11 @@ func NewSecureConnectProvider(configFilePath string) (Provider, error) {
 
 	file, err := os.ReadFile(configFilePath)
 	if err != nil {
-		if errors.Is(err, os.ErrNotExist) {
-			// Config file missing means Secure Connect is not supported.
-			return nil, errSourceUnavailable
-		}
-		return nil, err
+		// Config file missing means Secure Connect is not supported.
+		// There are non-os.ErrNotExist errors that may be returned.
+		// (e.g. if the home directory is /dev/null, *nix systems will
+		// return ENOTDIR instead of ENOENT)
+		return nil, errSourceUnavailable
```

Comment:
This could potentially break users who are looking for some other error, like permission denied.
It would be better if we could detect ENOTDIR somehow.

Reply:
I think this is okay for that case to fall back. This path happens in defaulting and I think it may be better to work in this case rather than now -- this sentinel is handled in a calling layer. I do understand though that hiding the details may make getting this feature working harder to detect. I think once we have logging in place in the near future details like this should be expressed in that way